parseRecipes.js

```diff
@@ -58,5 +62,19 @@ module.exports = function(recipeSlugs) {
       demo_index: recipe + '/index.html',
       intro_ref: recipe + '.html',
     };
+  }).sort(function(recipeA, recipeB) {
+    if (recipeA.category === recipeB.category) {
+      return recipeA.difficulty < recipeB.difficulty ? -1 : 1;
+    }
+
+    if (recipeA.category < recipeB.category) {
+      return -1;
+    }
+
+    return 1;
+  }).sort(function(recipeA, recipeB) {
+    // Sort recipes by preferred category
+    assert(categoryOrder.indexOf(recipeA.category) !== -1, recipeA.name + ': Unexpected category value "' + recipeA.category + '"');
```

Comment:
With `sort` some elements of the array could be missed. E.g. if you run this code:
```
[3, 4, 5].sort(function(a, b) {
  console.log(a);
  return a - b;
}
```
only *3* and *4* will be logged.

`forEach` would definitely be called for all the elements.